### PR TITLE
Add missing User Management API endpoints

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Changes

Adds all missing User endpoints:

- `Auth0::Api::V2::Users.get_roles`
- `Auth0::Api::V2::Users.remove_roles`
- `Auth0::Api::V2::Users.add_roles`
- `Auth0::Api::V2::Users.get_enrollments`
- `Auth0::Api::V2::Users.get_permissions`
- `Auth0::Api::V2::Users.remove_permissions`
- `Auth0::Api::V2::Users.add_permissions`
- `Auth0::Api::V2::Users.generate_recovery_code`
- `Auth0::Api::V2::Users.invalidate_browsers`

### Testing

Will add integration tests in another PR after Roles

* [x] This change adds unit test coverage
* [x] This change has been tested on the latest version of Ruby (2.6.0)

### Checklist

* [x] All existing and new tests complete without errors
* [x] Rubocop passes on all added/modified files
* [x] All active GitHub checks have passed

### CODE REVIEW

File: lib/auth0/api/v2/users.rb [hunk 1]
```diff
@@ -185,12 +185,137 @@ def user_logs(user_id, options = {})
...
+        def get_permissions(user_id)
+          raise Auth0::MissingUserId, 'Must supply a valid user_id' if user_id.to_s.empty?
+          get "#{users_path}/#{user_id}/permissions"
+        end
+
+        # Remove one or more permissions from a specific user.
+        # @see https://auth0.com/docs/api/management/v2#!/Users/delete_permissions
+        #
+        # @param user_id [string] The user_id of the permissions to remove.
+        # @param permissions [array] An array of Permissions to remove.
```
Comment (damieng) [inline lib/auth0/api/v2/users.rb]: Random uppercase P on permissions.

File: lib/auth0/api/v2/users.rb [hunk 2]
```diff
@@ -185,12 +185,137 @@ def user_logs(user_id, options = {})
...
+        #
+        # @param user_id [string] The user_id of the permissions to add.
+        # @param permissions [array] An array of Permissions to add.
+        def add_permissions(user_id, permissions)
+          raise Auth0::MissingUserId, 'Must supply a valid user_id' if user_id.to_s.empty?
+          permissions = validate_permissions_array permissions
+          post("#{users_path}/#{user_id}/permissions", permissions)
+        end
+
+        # Removes the current Guardian recovery code and generates and returns a new one.
```
Comment (damieng) [inline lib/auth0/api/v2/users.rb]: Should just be "remove" for consistency with other descriptions being mostly singular.

File: lib/auth0/api/v2/users.rb [hunk 3]
```diff
@@ -185,12 +185,137 @@ def user_logs(user_id, options = {})
...
+        # @see https://auth0.com/docs/api/management/v2#!/Users/post_recovery_code_regeneration
+        #
+        # @param user_id [string] The user_id of the recovery codes to regenerate.
+        def generate_recovery_code(user_id)
+          raise Auth0::MissingUserId, 'Must supply a valid user_id' if user_id.to_s.empty?
+          post "#{users_path}/#{user_id}/recovery-code-generation"
+        end
+
+        # Invalidates all remembered browsers for all authentication factors for a specific user.
```
Comment (damieng) [inline lib/auth0/api/v2/users.rb]: Invalidate

File: lib/auth0/api/v2/users.rb [hunk 4]
```diff
@@ -185,12 +185,137 @@ def user_logs(user_id, options = {})
...
+        # Check a roles array
+        def validate_roles_array(roles)
+          raise Auth0::InvalidParameter, 'Must supply an array of role names' unless roles.kind_of?(Array)
+          raise Auth0::InvalidParameter, 'Must supply an array of role names' if roles.empty?
+          raise Auth0::InvalidParameter, 'All role names must be strings' unless roles.all? {|role| role.is_a? String}
+        end
+
+        # Check a permissions array
+        def validate_permissions_array(permissions)
+          raise Auth0::InvalidParameter, 'Must supply an array of Permissions' unless permissions.kind_of?(Array)
```
Comment (damieng) [inline lib/auth0/api/v2/users.rb]: Capitalized Permissions here too should be permissions.

Comment (joshcanhelp) [reply lib/auth0/api/v2/users.rb]: 👆

File: lib/auth0/api/v2/users.rb (hunk 1 above)
Comment (joshcanhelp) [reply lib/auth0/api/v2/users.rb]: "Permissions" here refers to the struct I made for this PR.

**Pull request merged**